stacks_queues/queue.py

```diff
@@ -23,39 +23,60 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException()
+        if self.size == 0:
+            self.front = 0
+
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
+        self.store[self.rear] = element
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.size == 0:
+            raise QueueEmptyException()
+        if self.size == 1:
+            self.rear = -1
+        self.size -= 1
+        value = self.store[self.front]
+        self.store[self.front] = None
+        self.front = (self.front + 1) % self.buffer_size
+        return value
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.empty():
+            return None
+        return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        return self.size == 0
 
     def __str__(self):
         """ Returns the Queue in String form like:
             [3, 4, 7]
             Starting with the front of the Queue and
             ending with the rear of the Queue.
         """
-        pass
+        queue= []
+        for i in range(self.front, self. front + self.size):
+            i &= self.buffer_size
+            queue.append(self.store[i])
```

Comment:
This is causing an index out of range error.

```suggestion
        for i in range(self.front, self. front + self.size):
            index = i % self.buffer_size
            queue.append(self.store[index])
```